README.md

```diff
@@ -169,26 +172,46 @@ This section describes the Timer library Application Programming Interface.
   `static const bool IS_RECURRING = true`
 
 
-## SpinTimerAdapter
-* Adapter Interface, will notify `timeExpired()` event.
+### SpinTimerAction
+* Action Interface, will notify `timeExpired()` event.
 * Implementations derived from this interface can be injected into a SpinTimer object.
-  * the SpinTimer then will call out the specific adapter's `timeExpired()` method.
+  * the SpinTimer then will call out the specific action's `timeExpired()` method.
   Interface sending out a `timerExpired()` event.
-* *Time expired event*. To be implemented by specific `SpinTimerAdapter` class. `virtual void timeExpired() = 0`
+* *Time expired event*. To be implemented by specific `SpinTimerAction` class. `virtual void timeExpired() = 0`
 
-## UptimeInfoAdapter
+### UptimeInfoAdapter
 
 * Uptime Info Adapter Interface, will call out to `tMillis()` method to get current milliseconds counter value.
 * Implementations derived from this interface can be injected into the `UptimeInfo` singleton object.
 * Default implementation `DefaultUptimeInfoAdapter` for Arduino Framework environments is engaged automatically
 * Call out to get current milliseconds. To be implemented by specific `UptimeInfoAdapter` class. `virtual unsigned long tMillis() = 0`
 
+### Class diagram
+
+![SpinTimer Class Diagram](docs/pic/spintimer_class-diagram.bmp)
+
+[//]: # (\image html pic/spintimer_class-diagram.bmp)
+
+## Example Implementation
+
+This example shows how to use the SpinTimer library. The source code can be found in the Example folder.
+
+![Blink Example Class Diagram](docs/pic/spintimer_blink-example_class-diagram.bmp)
+
+[//]: # (\image html pic/spintimer_blink-example_class-diagram.bmp)
+
+![Blink Example Object Diagram](docs/pic/spintimer_blink-example_object-diagram.bmp)
+
+[//]: # (\image html pic/spintimer_blink-example_object-diagram.bmp)
+
+![Blink Example Sequence Diagram](docs/pic/spintimer_blink-example_sequence-diagram.bmp)
 
+[//]: # (\image html pic/spintimer_blink-example_sequence-diagram.bmp)
 
-# Notes
+## Notes
 This repository has been forked from  https://github.com/dniklaus/wiring-timer (Release 2.9.0) and with renamed Classes:
 * Timer -> SpinTimer
-* TimerAdapter -> SpinTimerAdapter
+* SpinTimerAdapter -> SpinTimerAction
```

Comment:
`SpinTimerAdapter` should really read as `TimerAdapter` here (because we reference to the old wiring-timer rel. 2.9.0 with the old naming)